src/vs/workbench/contrib/terminal/browser/terminal.ts

```diff
@@ -1045,6 +1045,11 @@ export interface IXtermTerminal {
 	 * Refreshes the terminal after it has been moved.
 	 */
 	refresh(): void;
+
+	/**
+	 * Gets the buffer content as a string
+	 */
+	getContent(startIndex?: number): string;
```

Comment:
I'm not sure this should live in `IXtermTerminal` as it's only used currently in `TerminalAccessibleBuffer`, it's also a little too specific to how that class works with the `startIndex` arg. We can move it in but I think we'd want to go with something more generic like this:

```ts
getWrappedLineIterator(start: number, end: number): IterableIterator<{ content: string, start: number, end: number }>
```

And then use that to simplify the implementation in `TerminalAccessibleBuffer`, encapsulating dealing with `isWrapped` in particular.

Reply:
yeah I wasn't sure what to call it, but makes sense

Reply:
hm maybe that can be deferred? would be good to get these fixes in. I wonder if we can just use 

https://github.com/microsoft/vscode/blob/a35b4b7281bdfc823e55bf485b7aa7421ebd800d/src/vs/workbench/contrib/terminal/browser/xterm/xtermTerminal.ts#L113-L121

Reply:
That's very similar but goes backwards, we could use that since we want new data which starts from the bottom.

Can you pull _getContent back into terminalAccessibleBuffer.ts first?